src/Calculator/Views/CalculatorProgrammerRadixOperators.xaml.cpp

```diff
@@ -47,7 +47,12 @@ void CalculatorProgrammerRadixOperators::FlyoutButton_Clicked(_In_ Platform::Obj
 
 void CalculatorProgrammerRadixOperators::checkDefaultBitShift()
 {
-    this->ArithmeticShiftButton->IsChecked = true;
```

Comment:
I don't think we need to call CollapseBitShiftButtons or set visibility here, it may cause a crash since some of those buttons may not be loaded yet at this point since they defer their loading. I think the only code we need here is this line:

`m_selectedShiftButtonMode = BitShiftMode::Arithmetic;`

Unless I am missing something else here that this is working around?

Reply:
This function is also called when the mode is changed in the calculator.
When the flyout menu is opened due to delaying initialization, the related button is initialized. but the menu does not open, then the related button is the same as the previous state

Reply:
Now I think about it, it just can be replaced to "ArithmeticShiftButton->IsChecked = true;" because it is called when mode changed in calc.
I'll check it

Reply:
Okay yeah you are right, I can see that when using the shortcut keys to access them. Can we factor out the logic to check if these buttons are loaded into a helper function and check that before calling CollapseBitShiftButtons just to double check that the buttons are initialized?

```    // Load deferred load buttons
    if (RolButton == nullptr)
    {
        FindName("RolButton");
        FindName("RorButton");
        FindName("RolCarryButton");
        FindName("RorCarryButton");
        FindName("LshLogicalButton");
        FindName("RshLogicalButton");
    }

    // Since arithmeticShiftButton defaults to IsChecked = true, this event an fire before we can load the deferred loaded controls. If that is the case, just
    // return and do nothing.
    if (RolButton == nullptr || RorButton == nullptr || RolCarryButton == nullptr || RorCarryButton == nullptr || LshLogicalButton == nullptr
        || RshLogicalButton == nullptr)
    {
        return;
    }

Reply:
as you said,
I added a check before calling Collapse Bit Shift Buttons.
However, the same code was duplicated in "checkDefaultBitShift" and "BitshiftFlyout_Checked". So I extracted as a function. Does it look okay?

Reply:
Perfect, thank you for the fix!